### PR TITLE
Enable override completion to use expression body accessors

### LINKED ISSUE #16334: IntelliSense: Cannot override properties with both accessors
**Version Used**: a2df65047264ed08fb3a985056085dbcee821253 

```C#
class B {
	public virtual int A { get; set; }
	class C : B {
		override A
	}
}
```

Press Ctrl+Space after the last `A`.

```
System.InvalidOperationException: Unexpected Null
   at Roslyn.Utilities.Contract.ThrowIfNull[T](T value, String message)
   at Microsoft.CodeAnalysis.Completion.Providers.AbstractMemberInsertingCompletionProvider.<DetermineNewDocumentAsync>d__8.MoveNext()
```

Demo: http://tryroslyn-edge.azurewebsites.net/#b:master/f:>ilr/MYGwhgzhAEBC0G8CwAoAkABwK4CMQEthoA3fAJwBcswRp8A7C6AQUWgHMBTCgbmgm58AvqjShIMAMLQAXHESi0Ae2KcyZfABNO0USJRCgA==

### PR DESCRIPTION
**Pull request by rchande**
Fixes #16334

### CODE REVIEW

**Comment by rchande** [comment]
Tag @dotnet/roslyn-ide for review

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/OverrideCompletionProviderTests_ExpressionBody.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        override A$$
+    }
+}";
+
+            var expectedCodeAfterCommit = @"class B
+{
+    public virtual int A { get; set; }
+    class C : B
+    {
+        public override int A { get => base.A$$; set => base.A = value; }
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/OverrideCompletionProviderTests_ExpressionBody.cs]: need test that just has a getter-only property.

**Comment by rchande** [comment]
**Customer scenario**

The customer has the C# code style option "Use expression bodied accessors" enabled and attempts to use IntelliSense to override a property. Instead of inserting the property definition, Visual Studio shows a dialog indicating that an exception occurred. 

**Bugs this fixes:** 

https://github.com/dotnet/roslyn/issues/16334

**Workarounds, if any**

In order to override properties with IntelliSense, users have to disable the code style option to use expression bodied accessors. This affects all other IDE features, and essentially prevents users from using expression bodied accessors.

**Risk**

Low. This adds a check to see if an accessor has a block or expression body.

**Performance impact**

Very low--the added extra check only runs when the user overrides an accessor.

**Is this a regression from a previous update?** 
No--this feature was never updated to handle the addition of expression bodied accessors.

**Root cause analysis:**

When testing the new code style/code generation options introduced to support expression bodied members, this particular feature was not tested.

**How was the bug found?**

Customer feedback

**Comment by Pilchie** [comment]
Tagging @MattGertz for escrow approval.

**Comment by CyrusNajmabadi** [comment]
👍

**Comment by natidea** [comment]
Approved to merge via [374658](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/374658)

**Pull request merged**